ocaml/xapi/recommendations.ml

```diff
@@ -0,0 +1,42 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module Unixext = Xapi_stdext_unix.Unixext
+module Config_file = Xcp_service.Config_file
+
+module D = Debug.Make (struct let name = "recommendations" end)
+
+open D
+module StringMap = Map.Make (String)
+
+let process_line map data =
+  match Config_file.parse_line data with
+  | Some (k, v) ->
+      debug "Parsing data, key: %s, value: %s" k v ;
+      StringMap.add k v map
+  | None ->
+      map
+
+let parse map filename =
+  debug "Parsing recommendations file: %s" filename ;
+  Unixext.file_lines_fold process_line map filename
+
+let load rec_dir =
+  (try Sys.readdir rec_dir with _ -> [||])
+  |> Array.to_list
+  |> List.stable_sort compare
+  |> List.map (Filename.concat rec_dir)
+  |> List.filter (fun f -> not (Sys.is_directory f))
```

Comment:
This is probably good enough in practice, but: you would rather have a filter that identifies regular files. Because this would still let devices and other non-regular files pass that are not directories.